db/migrate/20190805204523_drop_deprecated_columns.rb

```diff
@@ -0,0 +1,47 @@
+class DropDeprecatedColumns < ActiveRecord::Migration[5.1]
+  def change
+    remove_column :pipeline_run_stages, :failed_jobs
+
+    remove_column :host_genomes, :skip_deutero_filter
+
+    remove_column :metadata_fields, :validation_type
+
+    remove_column :pipeline_runs, :version
+    remove_column :pipeline_runs, :ready_step
```

Comment:
@boris-dimitrov , similarly, this blame to #862 which you reviewed. the col was last updated 2018-06-09. should we drop it?

Reply:
I'll let Boris comment, but regardless of the values, removing columns will cause pages to break if the ruby code references them, even if the column no longer serves a real purpose, e.g. https://github.com/chanzuckerberg/idseq-web/blob/df35dc7ec2c2ef2e84ade08efd4d096ae005b2d4/app/helpers/samples_helper.rb#L253

So I recommend following your investigation on the NULLs up with a comprehensive evaluation of all the code paths in which the column is referenced

Reply:
I would ask @davidrissato comment since, if I recall correctly, David recently removed the functionality to rerun from the failed stage, and now everything reruns from top.   Without this functionality, there may be no need for this particular column.  This doesn't mean, however, that the column is safe to remove.  It's entirely possible the state machine logic is needlessly entangled with that column in a way that could be time consuming to disentangle.

In general, for columns pertaining to the execution state machine, my suggestion would be to refrain from making any changes to this needlessly complex piece of machinery until such a time as someone can carefully examine, simplify, and re-test it all.   I do believe that would be a valuable project.

Reply:
> no need for this particular column

So this column was only intended for the "rerun stage" feature of admins? Then I think we should remove it assuming all references to it are deleted.

Reply:
I believe (at least part of) the intent was to mark from which point onwards a report could be shown. This used to be after stage 2. Now it's after stage 3. It was like a flag that could be passed to the frontend to decide between "not ready" and "ready" views.

Reply:
@cdebourcy  Do you think we should drop the column?

Reply:
@gregdingle I'm fine with dropping `ready_step` (and removing references to it) since it's easy enough to add back if someone wants to use it